src/dlsproto/node/neo/request/Put.d

```diff
@@ -12,77 +12,70 @@
 
 module dlsproto.node.neo.request.Put;
 
-public abstract scope class PutProtocol_v0
+import swarm.neo.node.IRequestHandler;
+
+public abstract class PutProtocol_v0: IRequestHandler
 {
+    import dlsproto.node.neo.request.core.Mixins;
+    import swarm.neo.connection.RequestOnConnBase;
+
     import swarm.neo.node.RequestOnConn;
     import dlsproto.common.Put;
     import ocean.transition;
     import core.stdc.time;
 
     /***************************************************************************
 
-        Request-on-conn, to get the event dispatcher and control the fiber.
+        Mixin the initialiser and the connection and resources members.
 
     ***************************************************************************/
 
-    protected RequestOnConn connection;
+    mixin IRequestHandlerRequestCore!();
+
+    /// Response status code to send to client.
+    private RequestStatusCode response;
 
     /**************************************************************************
 
         Request handler
 
         Params:
-            connection = connection to the client
             msg_payload = initial message read from client to begin
                 the request (the request code and version are
                 assumed to be extracted)
 
     **************************************************************************/
 
-    final public void handle ( RequestOnConn connection,
-            Const!(void)[] msg_payload )
+    public void preSupportedCodeSent ( Const!(void)[] msg_payload )
     {
-        this.connection = connection;
-        auto ed = this.connection.event_dispatcher;
-        auto parser = ed.message_parser;
-
         // Extract the channel, record's timestamp and value from the
         // message payload
-        cstring channel = parser.getArray!(char)(msg_payload);
-        time_t timestamp = *parser.getValue!(time_t)(msg_payload);
-        auto value = parser.getArray!(char)(msg_payload);
+        cstring channel = this.ed.message_parser.getArray!(char)(msg_payload);
```

Comment:
Btw David recommended using `parseBody`. See https://github.com/sociomantic-tsunami/dhtproto/pull/110#discussion_r187033167.

Reply:
Nice tip, I forgot that this was there.